experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -149,314 +148,662 @@ describe("LazyOptionalField", () => {
+			assert(primitiveOptionalField.is(SchemaBuilder.fieldOptional(leafDomain.number))); // TODO: this is presumably wrong
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldValue()));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldValue(Any)));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldValue(leafDomain.boolean)));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldValue(leafDomain.number)));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldSequence()));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldSequence(Any)));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldSequence(leafDomain.boolean)));
+			assert(!primitiveOptionalField.is(SchemaBuilder.fieldSequence(leafDomain.number)));
+			assert(
+				!primitiveOptionalField.is(
+					SchemaBuilder.fieldRecursive(FieldKinds.value, recursiveStructSchema),
 				),
+			);
+			assert(primitiveOptionalField.is(SchemaBuilder.fieldOptional()));
+
+			// #endregion
+		});
+
+		describe("at", () => {
+			it("Unboxes", () => {
+				const builder = new SchemaBuilder("test", undefined, leafDomain.library);
+				const rootSchema = SchemaBuilder.fieldOptional(leafDomain.number);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const { context, cursor } = initializeTreeWithContent({ schema, initialTree: 42 });
+
+				const field = new LazyOptionalField(context, rootSchema, cursor, rootFieldAnchor);
+
+				assert.equal(field.at(0), 42);
 			});
-			const rootSchema = SchemaBuilder.fieldOptional(builder.struct("struct", {}));
+		});
+
+		it("boxedAt", () => {
+			const builder = new SchemaBuilder("test", undefined, leafDomain.library);
+			const rootSchema = SchemaBuilder.fieldOptional(leafDomain.string);
 			const schema = builder.intoDocumentSchema(rootSchema);
 
-			const { context, cursor } = initializeTreeWithContent({ schema, initialTree: {} });
+			const { context, cursor } = initializeTreeWithContent({
+				schema,
+				initialTree: "Hello world",
+			});
 
-			// #endregion
+			const field = new LazyOptionalField(context, rootSchema, cursor, rootFieldAnchor);
+
+			const boxedResult = field.boxedAt(0);
+			assert.equal(boxedResult.type, leafDomain.string.name);
+			assert.equal(boxedResult.value, "Hello world");
+		});
+
+		it("parent", () => {
+			const builder = new SchemaBuilder("test", undefined, leafDomain.library);
+			const struct = builder.struct("struct", {
+				foo: SchemaBuilder.fieldOptional(...leafDomain.primitives),
+			});
+			const rootSchema = SchemaBuilder.fieldOptional(struct);
+			const schema = builder.intoDocumentSchema(rootSchema);
+
+			const { context, cursor } = initializeTreeWithContent({
+				schema,
+				initialTree: {
+					foo: "Hello world",
+				},
+			});
+
+			const rootField = new LazyOptionalField(context, rootSchema, cursor, rootFieldAnchor);
+			assert.equal(rootField.parent, undefined);
+
+			const parentPath: UpPath = {
```

Comment:
@CraigMacomber @yann-achard-MS Is this the best way to create an anchor to use as the child field's parent for this kind of test? Interested in thoughts on best patterns for testing at this layer.

Reply:
seems fine to me.